File: src/data/courses/dsa/dynamic-programming/classic-dp/egg-dropping/code/java/EggDropping.java
class Solution {
    public int superEggDrop(int K, int N) {
        int[][] dp=new int[N+1][K+1]; int t=0;
        while(dp[t][K]<N){t++;for(int k=1;k<=K;k++)dp[t][k]=1+dp[t-1][k-1]+dp[t-1][k];}
        return t;
    }
}

File: src/data/courses/dsa/dynamic-programming/classic-dp/egg-dropping/code/python/egg_dropping.py
class Solution:
    def superEggDrop(self, K, N) -> int:
        dp = [[0]*(K+1) for _ in range(N+1)]
        t = 0
        while dp[t][K] < N:
            t += 1
            for k in range(1, K+1): dp[t][k] = 1 + dp[t-1][k-1] + dp[t-1][k]
        return t
